definitions/scenarios/restore.rb

```diff
@@ -6,19 +6,27 @@ class Restore < ForemanMaintain::Scenario
       description 'Restore backup'
       param :backup_dir, 'Path to backup directory'
       param :incremental_backup, 'Is the backup incremental?'
+      param :dry_run, 'Check if backup could be restored, without performing the restore'
       manual_detection
     end
 
     # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
     def compose
       backup = ForemanMaintain::Utils::Backup.new(context.get(:backup_dir))
 
       add_steps(find_checks(:root_user))
       supported_version_check
       add_steps_with_context(Checks::Restore::ValidateBackup,
-                             Procedures::Restore::Confirmation,
```

Comment:
The confirmation should really happen after *all* checks, this is unrelated to the `dry-run` case I think (can split out in a separate PR, if you want to?)

Reply:
Yep, confirmation after _all checks_ would be nice, +1 for separate pr.